tests/service/views/test_dataset_views.py

```diff
@@ -926,3 +931,57 @@ def test_add_remote_and_local_file(svc_client_with_repo):
             assert pair[0].pop('job_id')
 
         assert set(pair[0].values()) == set(pair[1].values())
+
+
+@pytest.mark.service
+@pytest.mark.integration
+@flaky(max_runs=1, min_passes=1)
+def test_edit_datasets_view(svc_client_with_repo):
+    """Test editing dataset metadata."""
+    svc_client, headers, project_id, _ = svc_client_with_repo
+    short_name = '{0}'.format(uuid.uuid4().hex)
+
+    payload = {
+        'project_id': project_id,
+        'dataset_name': short_name,
+    }
+
+    response = svc_client.post(
+        '/datasets.create',
+        data=json.dumps(payload),
+        headers=headers,
+    )
+
+    assert response
+
+    assert_rpc_response(response)
+    assert {'dataset_name'} == set(response.json['result'].keys())
+    assert payload['dataset_name'] == response.json['result']['dataset_name']
+
+    params_list = {
+        'project_id': project_id,
+    }
+
+    response = svc_client.get(
+        '/datasets.list',
+        query_string=params_list,
+        headers=headers,
+    )
+
+    assert response
+    assert_rpc_response(response)
+
+    edit_payload = {
+        'project_id': project_id,
+        'short_name': short_name,
+        'title': 'my new title',
+    }
+    response = svc_client.post(
+        '/datasets.edit', data=json.dumps(edit_payload), headers=headers
+    )
+
+    assert response
+    assert_rpc_response(response)
+
+    assert {'warnings', 'edited'} == set(response.json['result'])
+    assert ['title'] == response.json['result']['edited']
```

Comment:
I am not sure how easy it is, but it would be good to also verify if the new title is correctly set.